### PR TITLE
Improve Dockerfile build process

### PR DESCRIPTION
**Pull request by damacus**
## Issues Resolved 

- [OPGOPS-1922](https://opgtransform.atlassian.net/browse/OPGOPS-1922)
- Partially [OPGOPS-2232](https://opgtransform.atlassian.net/browse/OPGOPS-2232)

## Description

Update the Docker builds to reduce image layers and layer sizes. 

## Improvements

- Remove php-fpm stats collector
- Add inspec testing into the containers we build. This we required since I collapsed a lot of layers into one.
- Collapse layers by having a single COPY command `COPY docker/. /` which copies the contents of the folder to the relative path on disc. 
This helps the users easily visualize where files are getting copied to without needing to read through 10+ COPY commands. 
- Update the Java 8 version to one that is available.

The tests mentioned in the comment below are run via inspec & docker-compose `make rebuild && make test`

### CODE REVIEW

**Comment by damacus** [comment]
✔  File /scripts/base should exist
  ✔  contabs: File /etc/crontab
     ✔  File /etc/crontab should exist
     ✔  File /var/spool/cron/crontabs/root should exist
     ✔  File /var/spool/cron/crontabs/app should exist


Profile Summary: 3 successful controls, 0 control failures, 0 controls skipped
Test Summary: 10 successful, 0 failures, 0 skipped
docker-compose -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Elasticsearch Shared Data (OPG Elasticsearch Shared Data)
Version: 0.1.0
Target:  docker://77062606b100379a7a6aeaf879c1b770d16374d5abcd8a6b133a67b88d173afb

  ✔  Elastic Folders: Elastic
     ✔  Directory /tmp/elasticsearchshareddata/beats/index-pattern should exist
     ✔  Directory /tmp/elasticsearchshareddata/beats/search should exist
     ✔  Directory /tmp/elasticsearchshareddata/beats/visualization should exist


Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 3 successful, 0 failures, 0 skipped
docker-compose -f tests/docker-compose.yml  -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Elastic5 (OPG Elastic5)
Version: 0.1.0
Target:  docker://0339029fe437f14775e89c58f5e8b533ae30868cf4811596c0c3495152136e6a

  ✔  ElasticSearch: ES
     ✔  Service elasticsearch should be installed
     ✔  Service elasticsearch should be enabled
  ✔  conf.d: conf.d
     ✔  File /usr/share/elasticsearch/config/elasticsearch.yml should exist
     ✔  File /usr/share/elasticsearch/config/log4j2.properties should exist


Profile Summary: 2 successful controls, 0 control failures, 0 controls skipped
Test Summary: 4 successful, 0 failures, 0 skipped
docker-compose -f tests/docker-compose.yml -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Base (OPG Base)
Version: 0.1.0
Target:  docker://05907c4ee5e4007f529f53393a40b94571543232095c042e9fab2c0f7ca660ed

  ✔  Java: Java
     ✔  Apt Repository https://download.docker.com/linux/ubuntu should exist
     ✔  Apt Repository https://download.docker.com/linux/ubuntu should be enabled
  ✔  Files: confd template
     ✔  File /srv/jenkins/.ssh/authorized_keys should exist


Profile Summary: 2 successful controls, 0 control failures, 0 controls skipped
Test Summary: 3 successful, 0 failures, 0 skipped
docker-compose -f tests/docker-compose.yml  -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Base (OPG Base)
Version: 0.1.0
Target:  docker://71ab48353aa5eb76b41cb5cb2ecbd6c7b5faa14f7448807be5267b7d3c310ea0

  ✔  Docker: Docker Repo
     ✔  Apt Repository https://download.docker.com/linux/ubuntu should exist
     ✔  Apt Repository https://download.docker.com/linux/ubuntu should be enabled
  ✔  Jenkins: ES
     ✔  Service jenkins should be installed
     ✔  Service jenkins should be enabled


Profile Summary: 2 successful controls, 0 control failures, 0 controls skipped
Test Summary: 4 successful, 0 failures, 0 skipped
docker-compose -f tests/docker-compose.yml  -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Base (OPG Base)
Version: 0.1.0
Target:  docker://40b752939524ebd42effacaba0038b6db5a2396c90f72b517ff302c7300bbe6f

  ✔  Java: Java
     ✔  Command java -version stderr should match /java version "1.8/


Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 1 successful, 0 failures, 0 skipped
docker-compose -f tests/docker-compose.yml  -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Elasticsearch (OPG Elasticsearch)
Version: 0.1.0
Target:  docker://dd0b1fef5f37fb0cfb50102c58ce91ebc94886416cb306e099fd310a690430a5

  ✔  Elastic Repo: Elastic Repo Setup
     ✔  Apt Repository https://artifacts.elastic.co/packages/5.x/apt should exist
     ✔  Apt Repository https://artifacts.elastic.co/packages/5.x/apt should be enabled
  ✔  Services: Kibana service
     ✔  Service kibana should be enabled
     ✔  Service kibana should be running
  ✔  Environment Variables: Environment Variables
     ✔  Environment variable OPG_SERVICE content should eq "kibana"


Profile Summary: 3 successful controls, 0 control failures, 0 controls skipped
Test Summary: 5 successful, 0 failures, 0 skipped
docker-compose -f tests/docker-compose.yml  -p test down
docker-compose -f tests/docker-compose.yml -p test up -d basegen
inspec exec tests -t docker://test_basegen_1

Profile: OPG Elasticsearch (OPG Elasticsearch)
Version: 0.1.0
Target:  docker://3a97f079accccdf100b8835d238b8f86bc8e051df8ff3de55ec9e65d8c8546eb

  ✔  MongoDB Repo: Repo Setup
     ✔  Apt Repository http://repo.mongodb.org/apt/ubuntu should exist
     ✔  Apt Repository http://repo.mongodb.org/apt/ubuntu should be enabled
  ✔  Services: MongoD service
     ✔  Service mongod should be enabled
     ✔  Service mongod should be running
  ✔  Backup Ninja: Backup Ninja Logging
     ✔  File /etc/backupninja.conf content should match /\/var\/log\/backupninja.log/
  ✔  Environment Variables: Environment Variables
     ✔  Environment variable OPG_SERVICE content should eq "mongodb"
     ✔  Environment variable MONGODB_VERSION content should eq "3.0.15"


Profile Summary: 4 successful controls, 0 control failures, 0 controls skipped
Test Summary: 7 successful, 0 failures, 0 skipped

File: Jenkinsfile-2 [hunk 1]
```diff
@@ -0,0 +1,59 @@
+def make_command(String, folder) {
```
Comment (carinadigital) [inline Jenkinsfile-2]: Why not just replace the existing jenkinsfile? It'll pick it up and run it in the branch run (but not the PRs)

Comment (damacus) [reply Jenkinsfile-2]: I probably will if I get it right, if not, I'll throw it away. 

I've no idea what the tag stag is actually doing.

**Comment by damacus** [comment]
Note to self, $RAND doesn't work in Jenkins

**Pull request closed without merging**